src/go/kreq-gen/kafka-request-generator.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"hash/crc32"
+	"math"
+	"math/rand"
+	"os"
+	"reflect"
+	"time"
+	"unsafe"
+
+	"github.com/twmb/franz-go/pkg/kmsg"
+)
+
+var crc32c = crc32.MakeTable(crc32.Castagnoli)
+
+func makeRandomBool() bool {
+	return rand.Intn(2) == 1
+}
+
+func makeRandomSign(n int64) int64 {
+	if makeRandomBool() {
+		return n * -1
+	}
+	return n
+}
+
+func makeRandomInt64n(bounds int64) int64 {
+	return makeRandomSign(rand.Int63n(bounds))
+}
+
+func makeRandomInt64() int64 {
+	return makeRandomInt64n(math.MaxInt64)
+}
+
+func makeRandomString() string {
+	letters := []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	b := make([]rune, rand.Intn(50))
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+func makeRandomTags() kmsg.Tags {
+	var tags kmsg.Tags
+	n := uint32(rand.Intn(10))
+	for i := uint32(0); i < n; i++ {
+		/// Make random bytes
+		token := make([]byte, rand.Intn(50))
+		rand.Read(token)
+		/// Choosing 100 as an arbitrary number so the tag IDs of unknown tags
+		/// never clash with those of known tags. Upon performing an audit the
+		/// largest tag id is < 10
+		tags.Set(100+i, token)
```

Comment:
I think the boundary for a varint switching from len 1 to len 2 is 127, so this may be good to start at 120